src/Error/TypeError.hs

```diff
@@ -0,0 +1,34 @@
+{-|
```

Comment:
I chose this module organization for two reasons:
1. It moves us closer to resolving #97 
2. Having a parent `Error` module and then children `(Type | Parse | Runtime)Error` modules matches the `Error` data type structure

Reply:
> It moves us closer to resolving #97

100% agree with this. Just getting the error structure out into it's own file is a great start.

> Having a parent Error module and then children (Type | Parse | Runtime)Error modules matches the Error data type structure

Agree with this as well